### PR TITLE
Add app.midterm.MidtermDesktop

### PR DESCRIPTION
**Pull request by minhloi**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by minhloi** [comment]
bot, build app.midterm.MidtermDesktop

**Comment by minhloi** [comment]
Can anyone help? I don't know what's wrong. I buildtthis on my computer just fine.

File: app.midterm.MidtermDesktop.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+app-id: app.midterm.MidtermDesktop
+runtime: org.freedesktop.Sdk
+runtime-version: "20.08"
+branch: stable
```
Comment (bilelmoussaoui) [inline app.midterm.MidtermDesktop.yml:4]:
```suggestion
```

File: app.midterm.MidtermDesktop.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: "20.08"
+command: midterm
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --filesystem=host
```
Comment (bilelmoussaoui) [inline app.midterm.MidtermDesktop.yml:14]: What does it need host access for?

File: app.midterm.MidtermDesktop.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      <image type="source">https://midterm.app/img/screenshots-linux/screenshot3-store.png</image>
+      <image type="source">https://midterm.app/img/screenshots-linux/screenshot4-store.png</image>
+      <image type="source">https://midterm.app/img/screenshots-linux/screenshot5-store.png</image>
+      <image type="source">https://midterm.app/img/screenshots-linux/screenshot6-store.png</image>
+    </screenshot>
+  </screenshots>
+  <releases>
+    <release version="0.9.2" date="2021-02-08"/>
+  </releases>
+  <content_rating type="oars-1.0" />
```
Comment (bilelmoussaoui) [inline app.midterm.MidtermDesktop.metainfo.xml:32]:
```suggestion
  <content_rating type="oars-1.1" />
```

File: app.midterm.MidtermDesktop.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>app.midterm.MidtermDesktop</id>
```
Comment (bilelmoussaoui) [inline app.midterm.MidtermDesktop.metainfo.xml:3]:
```suggestion
  <id>app.midterm.MidtermDesktop</id>
  <launchable type="desktop-id">app.midterm.MidtermDesktop.desktop</launchable>
```

File: app.midterm.MidtermDesktop.desktop [hunk 5]
```diff
@@ -0,0 +1,11 @@
+[Desktop Entry]
```
Comment (bilelmoussaoui) [inline app.midterm.MidtermDesktop.desktop:1]: Should be upstreamed along  with the metainfo, the icons as they are not Flathub specific.

File: app.midterm.MidtermDesktop.yml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      - install -Dm644 logo_512x512.png /app/share/icons/hicolor/512x512/apps/app.midterm.MidtermDesktop.png
+
+      - install -Dm644 app.midterm.MidtermDesktop.metainfo.xml /app/share/metainfo
+      - install -Dm644 app.midterm.MidtermDesktop.desktop -t /app/share/applications
+
+      # Install app wrapper
+      - install -D midterm /app/bin/
+    sources:
+      - type: archive
+        url: https://midterm-desktop-builds.s3.us-east-2.amazonaws.com/Midterm-0.9.2.tar.gz
```
Comment (bilelmoussaoui) [inline app.midterm.MidtermDesktop.yml:40]: any reason for not building from the source directly? :)

File: app.midterm.MidtermDesktop.yml (hunk 2 above)
Comment (minhloi) [reply app.midterm.MidtermDesktop.yml:16]: I tried to remove `--filesystem=host` but then I cannot pick images from file chooser so I think it's required.

File: app.midterm.MidtermDesktop.desktop (hunk 5 above)
Comment (minhloi) [reply app.midterm.MidtermDesktop.desktop]: I moved to the icons and the desktop file to the deb package. Thanks

**Comment by minhloi** [comment]
bot, build app.midterm.MidtermDesktop

**Comment by minhloi** [comment]
Why did the bot check for `builddir/*/share/app-info/xmls/app.midterm.MidtermDesktop.xml.gz` ?

How to solve this?

**Comment by minhloi** [comment]
bot, build app.midterm.MidtermDesktop

**Comment by minhloi** [comment]
bot, build app.midterm.MidtermDesktop

**Comment by minhloi** [comment]
Can you guys check if everything is good and merge this please? Thank you very much.

File: app.midterm.MidtermDesktop.yml (hunk 2 above)
Comment (nedrichards) [reply app.midterm.MidtermDesktop.yml:16]: Is there a more constrained set of folders than "anywhere on the computer" that would make sense? e.g. just the home directory or just the xdg-* folders for instance.

**Comment by minhloi** [comment]
bot, build app.midterm.MidtermDesktop

File: app.midterm.MidtermDesktop.yml (hunk 2 above)
Comment (minhloi) [reply app.midterm.MidtermDesktop.yml]: Thanks @nedrichards, I use `--filesystem=home` now. I think it's good enough.

**Pull request closed without merging**